Deprecate register-retry-count flag from controller-manager

gmarek opened the pull request:
I don't know if we have any established policy for flag removal.

cc @bgrant0607 @thockin

gmarek committed:
Remove register-retry-count flag from controller-manager

alex-mohr commented:
@davidopp for Controllers?

bgrant0607 commented:
We don't have a policy, but we need one.

Just ignore it and document it as deprecated. Otherwise, we break all turnup automation that used it.

bgrant0607 commented:
Also docs:

https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/admin/kube-controller-manager.md

I think we deleted the man pages, but there are copies in the wild:

http://rpm.pbone.net/index.php3/stat/45/idpl/29679766/numer/1/nazwa/kube-controller-manager

gmarek commented:
Changed to the deprecation notice. This flag will not show in --help anymore and will print out a warning if set. I already removed it from our .md file in a previous PR.

bgrant0607 commented on cmd/kube-controller-manager/app/controllermanager.go:
```diff
@@ -141,6 +141,8 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.Kubeconfig, "kubeconfig", s.Kubeconfig, "Path to kubeconfig file with authorization and master location information.")
 	fs.StringVar(&s.RootCAFile, "root-ca-file", s.RootCAFile, "If set, this root certificate authority will be included in service account's token secret. This must be a valid PEM-encoded CA bundle.")
+
+	fs.MarkDeprecated("register-retry-count", "This flag is currenty no-op and will be deleted.")
```
Could you please put this next to the flag definition?

davidopp commented:
Rassigned to @bgrant0607

bgrant0607 commented:
LGTM
